packages/sdk/tasks/deposit-bnb.ts

```diff
@@ -0,0 +1,432 @@
+    const OptimismPortal = new hre.ethers.Contract(
+      contractAddrs.l1.OptimismPortal,
+      Artifact__OptimismPortal.abi,
+      signer
+    )
+
+    console.log(
+      `L1CrossDomainMessenger: ${contractAddrs.l1.L1CrossDomainMessenger}`
+    )
+
+    console.log(`L1StandardBridge: ${contractAddrs.l1.L1StandardBridge}`)
+    const L1StandardBridge = new hre.ethers.Contract(
+      contractAddrs.l1.L1StandardBridge,
+      Artifact__L1StandardBridge.abi,
+      signer
+    )
+
+    const L2OutputOracle = new hre.ethers.Contract(
+      contractAddrs.l1.L2OutputOracle,
+      Artifact__L2OutputOracle.abi,
+      signer
+    )
+
+    const L2ToL1MessagePasser = new hre.ethers.Contract(
+      predeploys.L2ToL1MessagePasser,
+      Artifact__L2ToL1MessagePasser.abi,
+      l2Signer
+    )
+
+    const L2StandardBridge = new hre.ethers.Contract(
+      predeploys.L2StandardBridge,
+      Artifact__L2StandardBridge.abi,
+      l2Signer
+    )
+
+    const params = await OptimismPortal.params()
+    console.log('Intial OptimismPortal.params:')
+    console.log(params)
+
+    const L1BobaToken = new Contract(
+      l1BobaTokenAddress,
+      Artifact_BOBA.abi,
+      l1Signer
+    )
+
+    const L2BnbToken = new Contract(
+      l2BnbTokenAddress,
+      Artifact_BOBA.abi,
+      l2Signer
+    )
+
+    console.log(`Approving BOBA for deposit`)
+    const approvalBobaTx = await L1BobaToken.approve(
+      L1StandardBridge.address,
+      hre.ethers.constants.MaxUint256
+    )
+    await approvalBobaTx.wait()
+    console.log('BOBA approved')
+
+    console.log('Depositing BOBA to L2')
+    const depositBobaTx = await L1StandardBridge.depositERC20(
+      L1BobaToken.address,
+      l2BobaTokenAddress,
+      utils.parseEther('10'),
+      99999,
+      '0x'
+    )
+    await depositBobaTx.wait()
+    console.log(`ERC20 deposited - ${depositBobaTx.hash}`)
+
+    console.log('Checking to make sure deposit was successful')
+    // Deposit might get reorged, wait and also log for reorgs.
+    let prevBlockHash: string = ''
+    for (let i = 0; i < 12; i++) {
+      const messageReceipt = await signer.provider!.getTransactionReceipt(
+        depositBobaTx.hash
+      )
+      if (messageReceipt.status !== 1) {
+        console.log(`Deposit failed, retrying...`)
+      }
+
+      // Wait for stability, we want some amount of time after any reorg
+      if (prevBlockHash !== '' && messageReceipt.blockHash !== prevBlockHash) {
+        console.log(
+          `Block hash changed from ${prevBlockHash} to ${messageReceipt.blockHash}`
+        )
+        i = 0
+      } else if (prevBlockHash !== '') {
+        console.log(`No reorg detected: ${i}`)
+      }
+
+      prevBlockHash = messageReceipt.blockHash
+      await sleep(1000)
+    }
+    console.log(`Deposit BOBA confirmed`)
+
+    const l2BobaBalance = await l2Provider.getBalance(l2Signer.address)
+    if (l2BobaBalance.lt(utils.parseEther('10'))) {
```

Comment:
I'm no typescript expert, but this seems like it's asserting that the balance is simply non-zero -- shouldn't we be checking for an explicit balance?